test/e2e/basic/backup-volume-info/base.go

```diff
@@ -0,0 +1,163 @@
+			fmt.Printf("Skip snapshot case %s for vsphere environment.\n", v.CaseBaseName)
+			Skip("Skip snapshot case due to vsphere environment doesn't cover the CSI test, and it doesn't have a Velero native snapshot plugin.")
+		}
+
+		if strings.Contains(v.VeleroCfg.Features, "EnableCSI") {
+			if strings.Contains(v.CaseBaseName, "native-snapshot") {
+				fmt.Printf("Skip native snapshot case %s when the CSI feature is enabled.\n", v.CaseBaseName)
+				Skip("Skip native snapshot case due to CSI feature is enabled.")
+			}
+		} else {
+			if strings.Contains(v.CaseBaseName, "csi") {
+				fmt.Printf("Skip CSI related case %s when the CSI feature is not enabled.\n", v.CaseBaseName)
+				Skip("Skip CSI cases due to CSI feature is not enabled.")
+			}
+		}
+	})
+
+	v.CaseBaseName = v.CaseBaseName + v.UUIDgen
+	v.BackupName = "backup-" + v.CaseBaseName
+	v.RestoreName = "restore-" + v.CaseBaseName
+	v.TimeoutDuration = 10 * time.Minute
+	v.NamespacesTotal = 1
+
+	v.VeleroCfg = VeleroCfg
+	v.Client = *v.VeleroCfg.ClientToInstallVelero
+	v.NSIncluded = &[]string{v.CaseBaseName}
+
+	v.TestMsg = &TestMSG{
+		Desc:      "Test backup's VolumeInfo metadata content",
+		Text:      "The VolumeInfo should be generated based on the backup type",
+		FailedMSG: "Failed to verify the backup VolumeInfo's content",
+	}
+
+	v.BackupArgs = []string{
+		"backup", "create", v.BackupName,
+		"--namespace", v.VeleroCfg.VeleroNamespace,
+		"--include-namespaces", v.CaseBaseName,
+		"--snapshot-volumes" + "=" + strconv.FormatBool(v.SnapshotVolumes),
+		"--default-volumes-to-fs-backup" + "=" + strconv.FormatBool(v.DefaultVolumesToFSBackup),
+		"--snapshot-move-data" + "=" + strconv.FormatBool(v.SnapshotMoveData),
+		"--wait",
+	}
+
+	v.RestoreArgs = []string{
+		"create", "--namespace", v.VeleroCfg.VeleroNamespace, "restore", v.RestoreName,
+		"--from-backup", v.BackupName, "--wait",
+	}
+	return nil
+}
+
+func (v *BackupVolumeInfo) CreateResources() error {
+	v.Ctx, v.CtxCancel = context.WithTimeout(context.Background(), v.TimeoutDuration)
+	labels := map[string]string{
+		"volume-info": "true",
+	}
+	for nsNum := 0; nsNum < v.NamespacesTotal; nsNum++ {
+		fmt.Printf("Creating namespaces ...\n")
+		createNSName := v.CaseBaseName
+		if err := CreateNamespaceWithLabel(v.Ctx, v.Client, createNSName, labels); err != nil {
+			return errors.Wrapf(err, "Failed to create namespace %s", createNSName)
+		}
+
+		// Install StorageClass
+		Expect(InstallTestStorageClasses(fmt.Sprintf("../testdata/storage-class/%s-csi.yaml", v.VeleroCfg.CloudProvider))).To(Succeed(), "Failed to install StorageClass")
+
+		// No need to install the VSC for vsphere environment.
+		if v.VeleroCfg.CloudProvider != "vsphere" {
+			// Install VolumeSnapshotClass
+			Expect(KubectlApplyByFile(v.Ctx, fmt.Sprintf("../testdata/volume-snapshot-class/%s.yaml", v.VeleroCfg.CloudProvider))).To(Succeed(), "Failed to install VolumeSnapshotClass")
```

Comment:
VolumeSnapshotClass installation should be moved to Velero installation part, because other test cases with CSI snapshot test steps will also need VolumeSnapshotClass if it is not installed before E2E execution. This test case can not guarantee CSI plugin is passed to the E2E test, currently that depends on CI to pass CSI plugin and EnableCSI flag to E2E test, so we assume we can not have native-snapshot and CSI-snapshot in the same test execution, it's not as design for now.

Reply:
Make sense.

Reply:
Modified.